cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/transformers/TwoWindingsTransformerConversion.java

```diff
@@ -137,4 +191,111 @@ private void setRegulatingControlContext(ConvertedT2xModel convertedT2xModel, Tw
+        if (tapChanger != null) {
+            return tapChanger.getSteps().get(getStepIndex(tapChanger)).getR();
+        }
+        return 0.0;
+    }
+
+    private static double getStepX(TapChanger tapChanger) {
+        if (tapChanger != null) {
+            return tapChanger.getSteps().get(getStepIndex(tapChanger)).getX();
+        }
+        return 0.0;
+    }
+
+    private static double getStepG1(TapChanger tapChanger) {
+        if (tapChanger != null) {
+            return tapChanger.getSteps().get(getStepIndex(tapChanger)).getG1();
+        }
+        return 0.0;
+    }
+
+    private static double getStepB1(TapChanger tapChanger) {
+        if (tapChanger != null) {
+            return tapChanger.getSteps().get(getStepIndex(tapChanger)).getB1();
+        }
+        return 0.0;
+    }
+
+    private static double getValue(double initialValue, double rtcStepValue, double ptcStepValue) {
+        return initialValue * (1 + rtcStepValue / 100) * (1 + ptcStepValue / 100);
+    }
+
+    private static double getR(ConvertedT2xModel convertedT2xModel) {
+        return getValue(convertedT2xModel.r, getStepR(convertedT2xModel.end1.ratioTapChanger), getStepR(convertedT2xModel.end1.phaseTapChanger));
+    }
+
+    private static double getX(ConvertedT2xModel convertedT2xModel) {
+        return getValue(convertedT2xModel.x, getStepX(convertedT2xModel.end1.ratioTapChanger), getStepX(convertedT2xModel.end1.phaseTapChanger));
+    }
+
+    private static double getG(ConvertedT2xModel convertedT2xModel) {
+        return getValue(convertedT2xModel.end1.g, getStepG1(convertedT2xModel.end1.ratioTapChanger), getStepG1(convertedT2xModel.end1.phaseTapChanger));
+    }
+
+    private static double getB(ConvertedT2xModel convertedT2xModel) {
+        return getValue(convertedT2xModel.end1.b, getStepB1(convertedT2xModel.end1.ratioTapChanger), getStepB1(convertedT2xModel.end1.phaseTapChanger));
+    }
+
+    private PiModel piModel(double r, double x, double g, double b, double a, double angle) {
+        PiModel piModel = new PiModel();
+
+        Complex ratio = ComplexUtils.polar2Complex(a, angle);
+        Complex ytr = new Complex(r, x).reciprocal();
+        Complex ytr1 = ytr.divide(ratio.conjugate());
+        Complex ytr2 = ytr.divide(ratio);
+        Complex ysh1 = new Complex(g, b).divide(ratio.multiply(ratio.conjugate()))
+            .add(ytr.multiply(ratio.reciprocal().subtract(1.0)).divide(ratio.conjugate()));
+        Complex ysh2 = ytr.multiply(ratio.reciprocal().negate().add(1.0));
+
+        piModel.r1 = ytr1.reciprocal().getReal();
+        piModel.x1 = ytr1.reciprocal().getImaginary();
+        piModel.g1 = ysh1.getReal();
+        piModel.b1 = ysh1.getImaginary();
+        piModel.r2 = ytr2.reciprocal().getReal();
+        piModel.x2 = ytr2.reciprocal().getImaginary();
+        piModel.g2 = ysh2.getReal();
+        piModel.b2 = ysh2.getImaginary();
+        return piModel;
+    }
+
+    private static class PiModel {
```

Comment:
I understand here that we need to build a pi model without any ratio. I am not familiar with this kind of pi model. But if I understand well, you will just keep the attributes side 1 to build the dangling line? In a LF computation on an IGM, we will have differences, but it will be more acceptable after merging with the other side, right?

Reply:
Indeed if you can add some comments to explicit the conversion, it could be useful for support. Note that most of the methods you use to build your pi model are identical as those we use in the open-loadflow. Maybe we have to think about putting them as utility functions later.

Reply:
I mapped the transformer to a danglingLine in a more clear way. We can map directly to it if we fix the ratio always to 1.0.
In the same way we can map the transformer to half line of the TieLine converting it to an equivalent branch if we fix the phase angle to 0.0. In the first case we will have differences in a LF computation if the ratio of the transformer is not 1.0. In the second case the differences will appear if the transformer has a phaseTapChanger and the phase angle is not zero. 
In the next step we will expand the danglingLine and the TieLine to better support the transformer. At the end, we will considerer putting all the needed methods as utility functions.